platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/ConfigurationAdminImpl.java

```diff
@@ -425,19 +383,42 @@ final boolean isAllowedPid(final String pid) {
         }
     }
 
-    public void setConfigurationAdminPluginList(
+    public String getDefaultFactoryLdapFilter() {
+        return "(" + SERVICE_FACTORYPID + "=" + "*)";
+    }
+
+    public String getDefaultLdapFilter() {
+        return "(" + SERVICE_PID + "=" + "*)";
+    }
+
+    void setConfigurationAdminPluginList(
             List<ConfigurationAdminPlugin> configurationAdminPluginList) {
         this.configurationAdminPluginList = configurationAdminPluginList;
     }
 
+    public Metatype findMetatypeForConfig(Configuration config) {
+        List<Service> services = listServices(getDefaultFactoryLdapFilter(),
+                getDefaultLdapFilter());
+        for (Service service : services) {
+            String id = String.valueOf(service.getId());
```

Comment:
❗️ Do we need `String.valueOf()` here since `service.getId()` already returns a `String`?